privacy-protections/storage-partitioning/helpers/commonTests.js

```diff
@@ -0,0 +1,552 @@
+    if (
+        (sameSites.every(v => v.error === 'Unsupported')) &&
+        (crossSites.every(v => v.error === 'Unsupported'))
+    ) {
+        return 'unsupported';
+    } else if (
+        (sameSites.every(v => v.error && v.error.endsWith('is deprecated'))) &&
+        (crossSites.every(v => v.error && v.error.endsWith('is deprecated')))
+    ) {
+        return 'unsupported';
+    }
+
+    if (reference.value !== random) {
+        if (reference.value === null && typeof reference.error !== 'undefined') {
+            return 'error';
+        }
+        return 'fail';
+    }
+
+    if (
+        // (!sameSites.length === configurations['same-site'].iterations) ||
+        // (!crossSites.length === configurations['cross-site'].iterations) ||
+        (!sameSites.every(v => v.value === reference.value)) ||
+        (!crossSites.every(v => v.value === crossSites[0].value)) ||
+        (!crossSites.every(v => v.value !== reference.value))
+    ) {
+        return 'fail';
+    }
+    return 'pass';
+}
+
+// Validates the results returned by Cache APIs. These return a count of the
+// number of requests the server has received upon each pageload. E.g.,
+//
+// Iframe Cache
+//     same-site: [ { "value": "1" }, { "value": "1" } ]
+//    cross-site: [ { "value": "2" }, { "value": "2" } ]
+function validateCacheAPI (sameSites, crossSites) {
+    if (
+        (sameSites.every(v => v.error === 'No requests received')) &&
+        (crossSites.every(v => v.error === 'No requests received'))
+    ) {
+        return 'unsupported';
+    } else if (
+        (sameSites.every(v => v.value === sameSites[0].value)) &&
+        (crossSites.every(v => v.value === crossSites[0].value)) &&
+        (crossSites.every(v => v.value === sameSites[0].value + 1))
+    ) {
+        return 'pass';
+    } else if (
+        (sameSites.every(v => v.value === sameSites[0].value)) &&
+        (crossSites.every(v => v.value === crossSites[0].value)) &&
+        (crossSites.every(v => v.value === sameSites[0].value))
+    ) {
+        return 'fail';
+    }
+    return 'error';
+}
+
+const testAPIs = {
+    'document.cookie': {
+        type: 'storage',
+        store: (data) => {
+            // we are using same set of tests for main frame and an iframe
+            // we want to set 'Lax' in the main frame for the cookie not to be suspicious
+            // we want to set 'None' in an iframe for cookie to be accessible to us
+            const sameSite = (window !== window.top) ? 'None' : 'Lax';
+
+            document.cookie = `jsdata=${data}; expires= Wed, 21 Aug 2030 20:00:00 UTC; Secure; SameSite=${sameSite}`;
+        },
+        retrieve: () => {
+            return document.cookie.match(/jsdata=([0-9a-z-]+)/)[1];
+        },
+        validate: validateStorageAPI
+    },
+    localStorage: {
+        type: 'storage',
+        store: (data) => {
+            localStorage.setItem('data', data);
```

Comment:
same here, we are already using 'data'. Maybe let's use something that includes test page name across the board?